tests/unit/modules/test_openscap.py

```diff
@@ -1,213 +1,245 @@
 from tests.support.unit import TestCase
 
 
-class OpenscapTestCase(TestCase):
-
-    random_temp_dir = "/tmp/unique-name"
-    policy_file = "/usr/share/openscap/policy-file-xccdf.xml"
-
-    def setUp(self):
-        import salt.modules.openscap
-
-        salt.modules.openscap.__salt__ = MagicMock()
-        patchers = [
-            patch("salt.modules.openscap.__salt__", MagicMock()),
-            patch("salt.modules.openscap.shutil.rmtree", Mock()),
-            patch(
-                "salt.modules.openscap.tempfile.mkdtemp",
-                Mock(return_value=self.random_temp_dir),
-            ),
-        ]
-        for patcher in patchers:
-            self.apply_patch(patcher)
-
-    def apply_patch(self, patcher):
-        patcher.start()
-        self.addCleanup(patcher.stop)
-
-    def test_openscap_xccdf_eval_success(self):
-        with patch(
-            "salt.modules.openscap.Popen",
-            MagicMock(
-                return_value=Mock(
-                    **{"returncode": 0, "communicate.return_value": ("", "")}
-                )
-            ),
+class OpenscapTestCase(TestCase, LoaderModuleMockMixin):
+    """
+    Test cases for salt.module.openscap
+    """
+
+    def setup_loader_modules(self):
+        return {openscap: {}}
+
+    def test_version(self):
+        """
+        Test, if the short version is returned.
+        """
+
+        mocked_oscap_string = """OpenSCAP command line tool (oscap) 1.2.16
+Copyright 2009--2017 Red Hat Inc., Durham, North Carolina.
+
+==== Supported specifications ====
+XCCDF Version: 1.2
+OVAL Version: 5.11.1
+
+==== Capabilities added by auto-loaded plugins ====
+SCE Version: 1.0 (from libopenscap_sce.so.8)
+
+==== Paths ====
+Schema files: /usr/share/openscap/schemas
+Default CPE files: /usr/share/openscap/cpe
+Probes: /usr/lib/x86_64-linux-gnu/openscap
+
+==== Inbuilt CPE names ====
+Red Hat Enterprise Linux - cpe:/o:redhat:enterprise_linux
+Red Hat Enterprise Linux 5 - cpe:/o:redhat:enterprise_linux:5
+
+==== Supported OVAL objects and associated OpenSCAP probes ====
+OVAL family   OVAL object                  OpenSCAP probe
+----------    ----------                   ----------
+(null)        system_info                  probe_system_info
+independent   family                       probe_family"""
+
+        with patch.dict(
+            openscap.__salt__, {"cmd.run": MagicMock(return_value=mocked_oscap_string)}
+        ), patch.object(
+            openscap, "_oscap_cmd", MagicMock(return_value="/usr/bin/oscap")
+        ):
+            # Test if short output is correct
+            self.assertEqual(openscap.version(), {"oscap": "1.2.16"})
+
+            # Test if long output is correct
+            self.assertIsInstance(openscap.version("all"), dict)
+
+    def test_build_cmd(self):
+        """
+        Test if _build_cmd returns well formed cmds.
+        """
+        with patch.object(
+            openscap, "_has_operation", MagicMock(return_value=True)
```

Comment:
**suggestion (non-blocking)**: generally prefer `autospec=True` or `create_autospec`. Not 100% of the time, but typically it's better, as it tracks changes in the underlying API. Throughout this PR it looks like rather than `MagicMock`, can do:

```
with patch.object(openscap, "_has_operation", autospec=True, return_value=True):
```
Or similar. For something like `cmd.run`, can do an 

```
import salt.modules.cmdmod as cmdmod
...
patch.dict(..., {"cmd.run": create_autospec(cmdmod.run, return_value="whatever")})
```
(or something to that effect. Just helps to ensure that tests will fail if the underlying API changes.

Reply:
I used  test_iptables.py module as a reference to get up to speed on the whole unittest (was my first time writing unitests). Do you want me to update mine or is it just a thing I should consider the next time?

Reply:
@waynew do you want me to update the Tests or cann I set it to resolved?